arbnode/seq_coordinator.go

```diff
@@ -663,8 +664,9 @@ func (c *SeqCoordinator) update(ctx context.Context) time.Duration {
 	}
 
 	// update wanting the lockout
+	// Sequencer should want lockout if and only if- its synced, not avoiding lockout and processedMessages is not lagging too much behind localMsgCount
 	var wantsLockoutErr error
-	if synced && !c.AvoidingLockout() {
+	if synced && !c.AvoidingLockout() && processedMessages+10 >= localMsgCount {
```

Comment:
I think I understood the problem - due to execution split "synced" now tests that the consensus is synced and not that execution is synced... nice!

Instead of an arbitrary 10 messages forward - do the same thing that execution sync monitor test:
check processedMessgeCount against syncMonitor.SyncTargetMessageCount()
It returns the message count from approximately 1 second ago - so this node is considered in sync if:
* consensus is in sync
* execution processed every message that consensus had 1 second ago

Reply:
Fixed it. Thanks for the suggestion